test/python/test_composite.py

```diff
@@ -1081,159 +1088,157 @@ def test_input_data_user_modifications(self):
         gas.update_user_data(extra)
         data2 = gas.input_data
-        self.assertEqual(extra["foo"], data2["foo"])
-        self.assertEqual(extra["bar"], data2["bar"])
+        assert extra["foo"] == data2["foo"]
+        assert extra["bar"] == data2["bar"]
         gas.clear_user_data()
         data3 = gas.input_data
-        self.assertEqual(data1, data3)
+        assert data1 == data3
 
     def test_input_data_state(self):
         gas = ct.Solution('h2o2.yaml', transport_model=None)
         data = gas.input_data
-        self.assertEqual(gas.T, data['state']['T'])
-        self.assertEqual(gas.density, data['state']['density'])
+        assert gas.T == data['state']['T']
+        assert gas.density == data['state']['density']
 
         gas.TP = 500, 3.14e5
         data = gas.input_data
-        self.assertEqual(gas.T, data['state']['T'])
-        self.assertEqual(gas.density, data['state']['density'])
+        assert gas.T == data['state']['T']
+        assert gas.density == data['state']['density']
 
     def test_input_data_custom(self):
         gas = ct.Solution('ideal-gas.yaml')
         data = gas.input_data
-        self.assertEqual(data['custom-field']['first'], True)
-        self.assertEqual(data['custom-field']['last'], [100, 200, 300])
+        assert data['custom-field']['first'] == True
+        assert data['custom-field']['last'] == [100, 200, 300]
 
         if sys.version_info >= (3,7):
             # Check that items are ordered as expected
-            self.assertEqual(
-                list(data),
-                ["name", "thermo", "elements", "species", "state",
-                 "custom-field", "literal-string"]
-            )
-            self.assertEqual(list(data["custom-field"]), ["first", "second", "last"])
-            self.assertEqual(data["literal-string"], "spam\nand\neggs\n")
+            assert list(data) == ["name", "thermo", "elements", "species", "state",
+                                  "custom-field", "literal-string"]
+            assert list(data["custom-field"]) == ["first", "second", "last"]
+            assert data["literal-string"] == "spam\nand\neggs\n"
 
     def test_input_data_debye_huckel(self):
         soln = ct.Solution('thermo-models.yaml', 'debye-huckel-B-dot-ak')
         data = soln.input_data
-        self.assertEqual(data['thermo'], 'Debye-Huckel')
+        assert data['thermo'] == 'Debye-Huckel'
         act_data = data['activity-data']
-        self.assertEqual(act_data['model'], 'B-dot-with-variable-a')
-        self.assertEqual(act_data['default-ionic-radius'], 4e-10)
-        self.assertNotIn('kinetics', data)
-        self.assertNotIn('transport', data)
+        assert act_data['model'] == 'B-dot-with-variable-a'
+        assert act_data['default-ionic-radius'] == 4e-10
+        assert 'kinetics' not in data
+        assert 'transport' not in data
 
-    def test_yaml_simple(self):
+    def test_yaml_simple(self, load_yaml):
```

Comment:
Could `load_yaml` be marked as an "autouse" fixture, removing the need to specify it as an argument?

Reply:
Yes. I originally was thinking that the "autouse" was something to be used sparingly as it seems like having too many of them might clutter things up possibly. I could be wrong, but I think the high level goal with the pytest fixtures is to use them as arguments primarily.

Reply:
Actually, the load_yaml doesn't have any side effects, so although the fixture is applied, it isn't injected automatically (as far as I understand). So we would need to pass it as an argument, versus say the other fixtures that apply to a class and adjust class attributes as a side effect.

Reply:
Oh, that's a good point -- this is hardly a fixture at all. `load_yaml` is just a pure function that could be defined in `utilities.py`.

Reply:
I moved load_yaml back to the utilities as a function that is loaded per-test-module.